src/modal/modal.js

```diff
@@ -261,18 +249,21 @@ angular.module('ui.bootstrap.modal', [])
       }
 
       function removeAfterAnimate(domEl, scope, done) {
-        // Closing animation
-        scope.animate = false;
-
-        if (domEl.attr('modal-animation') && $animate.enabled()) {
-          // transition out
-          domEl.one('$animate:close', function closeFn() {
-            $rootScope.$evalAsync(afterAnimating);
-          });
-        } else {
-          // Ensure this call is async
-          $timeout(afterAnimating);
-        }
+        var closingDeferred;
+        var closingPromise;
+        var setIsAsync = function () {
+          if (!closingDeferred) {
+            closingDeferred = $q.defer();
+            closingPromise = closingDeferred.promise;
+          }
+
+          return function () {
+            closingDeferred.resolve();
+          };
+        };
+        scope.$broadcast($modalStack.NOW_CLOSING_EVENT, setIsAsync);
+
+        return $q.when(closingPromise).then(afterAnimating);
```

Comment:
Yes, there is, if `setIsAsync` is never called after the `NOW_CLOSING_EVENT` has broadcasted, `closingPromise` will be `undefined`. This is intentional. In such a case, the promise should resolve immediately (asynchronously), as it has no need to wait for anything (animations etc.)